File: api/projects.js
const { createClient } = require('@supabase/supabase-js');

// Configuração do Supabase
const supabaseUrl = 'https://qngnbyueqdewjjzgbkun.supabase.co';
const supabaseKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InFuZ25ieXVlcWRld2pqemdia3VuIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTYxMjE0MjYsImV4cCI6MjA3MTY5NzQyNn0.F-OOhtVHz-c2FiGMkl6e3lbzhvvG5uo5SC2z58_FSIY';
const supabase = createClient(supabaseUrl, supabaseKey);

module.exports = async (req, res) => {
  // CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization');
  
  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  try {
    const { method, query, body, url } = req;
    const { action, id } = query;
    
    // Extrair ID da URL para requisições DELETE/PUT
    let projectId = id;
    if (!projectId && url) {
      const urlParts = url.split('/');
      const lastPart = urlParts[urlParts.length - 1];
      if (lastPart && !isNaN(lastPart)) {
        projectId = lastPart;
      }
    }

    if (method === 'GET') {
      if (projectId) {
        // Buscar projeto específico
        const { data, error } = await supabase
          .from('projects')
          .select('*')
          .eq('id', projectId)
          .single();
        
        if (error) throw error;
        return res.json({ success: true, data });
      } else if (action === 'answers' && query.id) {
        // Buscar respostas do projeto
        const { data, error } = await supabase
          .from('project_product_answers')
          .select('*')
          .eq('project_id', query.id);
        
        if (error) throw error;
        return res.json({ success: true, data: data || [] });
      } else {
        // Buscar todos os projetos
        const { data, error } = await supabase
          .from('projects')
          .select('*')
          .order('created_at', { ascending: false });
        
        if (error) throw error;
        return res.json({ success: true, data: data || [] });
      }
    }

    if (method === 'POST') {
      if (action === 'save-answers') {
        // Salvar respostas do projeto
        const { project_id, answers } = body;
        
        // Primeiro, deletar respostas existentes
        await supabase
          .from('project_product_answers')
          .delete()
          .eq('project_id', project_id);
        
        // Inserir novas respostas
        if (answers && answers.length > 0) {
          const { error } = await supabase
            .from('project_product_answers')
            .insert(answers);
          
          if (error) throw error;
        }
        
        return res.json({ success: true, message: 'Respostas salvas com sucesso' });
      } else {
        // Criar novo projeto
        console.log('🔢 Projeto padrão - valor original do frontend:', body.totalScopeHours);
        
        let projectData = { ...body };
        
        // Para projetos customizados, arredondar as horas
        if (body.isCustom && body.totalScopeHours) {
          projectData.totalScopeHours = Math.ceil(body.totalScopeHours);
          console.log('🔄 Projeto customizado - arredondando horas:', projectData.totalScopeHours);
        } else {
          console.log('🔍 Projeto padrão - usando horas calculadas no frontend');
          console.log('⏰ Horas totais do frontend:', body.totalScopeHours);
        }
        
        const { data, error } = await supabase
          .from('projects')
          .insert([projectData])
          .select()
          .single();
        
        if (error) throw error;
        
        return res.json({ success: true, data, message: 'Projeto criado com sucesso' });
      }
    }

    if (method === 'PUT') {
      // Atualizar projeto
      const updateProjectId = projectId || body.id;
      if (!updateProjectId) {
        return res.status(400).json({ success: false, error: 'ID do projeto é obrigatório' });
      }
      
      const { data, error } = await supabase
        .from('projects')
        .update(body)
        .eq('id', updateProjectId)
        .select()
        .single();
      
      if (error) throw error;
      
      return res.json({ success: true, data, message: 'Projeto atualizado com sucesso' });
    }

    if (method === 'DELETE') {
      // Excluir projeto
      if (!projectId) {
        return res.status(400).json({ success: false, error: 'ID do projeto é obrigatório' });
      }
      
      // Primeiro, excluir os itens de escopo do projeto
      await supabase
        .from('project_scope_items')
        .delete()
        .eq('project_id', projectId);
      
      // Depois, excluir o projeto
      const { error } = await supabase
        .from('projects')
        .delete()
        .eq('id', projectId);
      
      if (error) throw error;
      
      return res.json({ success: true, message: 'Projeto excluído com sucesso' });
    }

    return res.status(405).json({ error: 'Método não permitido' });
  } catch (error) {
    console.error('Erro na API de projetos:', error);
    return res.status(500).json({ success: false, error: error.message });
  }
};

File: api/projects.py
import json
from supabase import create_client, Client
from urllib.parse import parse_qs, urlparse

# Configuração do Supabase
SUPABASE_URL = "https://qngnbyueqdewjjzgbkun.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InFuZ25ieXVlcWRld2pqemdia3VuIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTYxMjE0MjYsImV4cCI6MjA3MTY5NzQyNn0.F-OOhtVHz-c2FiGMkl6e3lbzhvvG5uo5SC2z58_FSIY"

try:
    supabase: Client = create_client(SUPABASE_URL, SUPABASE_KEY)
except Exception as e:
    print(f"Erro ao conectar com Supabase: {e}")
    supabase = None

def handler(request):
    # Headers CORS
    headers = {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
        'Access-Control-Allow-Headers': 'Content-Type',
        'Content-Type': 'application/json'
    }
    
    # Handle OPTIONS request (CORS preflight)
    if request.method == 'OPTIONS':
        return {
            'statusCode': 200,
            'headers': headers,
            'body': ''
        }
    
    try:
        if not supabase:
            return {
                'statusCode': 500,
                'headers': headers,
                'body': json.dumps({"error": "Erro de conexão com banco"})
            }

        # Parse URL para obter parâmetros
        url_parts = urlparse(request.url)
        path_parts = url_parts.path.split('/')
        query_params = parse_qs(url_parts.query)
        
        if request.method == 'GET':
            # Se há ID na URL, buscar projeto específico
            if len(path_parts) > 3 and path_parts[3]:
                project_id = path_parts[3]
                response = supabase.table('projects').select('*').eq('id', project_id).execute()
            else:
                # Buscar todos os projetos
                response = supabase.table('projects').select('*').execute()
            
            return {
                'statusCode': 200,
                'headers': headers,
                'body': json.dumps(response.data)
            }
            
        elif request.method == 'POST':
            body = json.loads(request.body)
            response = supabase.table('projects').insert(body).execute()
            return {
                'statusCode': 201,
                'headers': headers,
                'body': json.dumps(response.data)
            }
            
        elif request.method == 'PUT':
            if len(path_parts) > 3 and path_parts[3]:
                project_id = path_parts[3]
                body = json.loads(request.body)
                response = supabase.table('projects').update(body).eq('id', project_id).execute()
                return {
                    'statusCode': 200,
                    'headers': headers,
                    'body': json.dumps(response.data)
                }
            else:
                return {
                    'statusCode': 400,
                    'headers': headers,
                    'body': json.dumps({"error": "ID do projeto é obrigatório para atualização"})
                }
                
        elif request.method == 'DELETE':
            if len(path_parts) > 3 and path_parts[3]:
                project_id = path_parts[3]
                response = supabase.table('projects').delete().eq('id', project_id).execute()
                return {
                    'statusCode': 200,
                    'headers': headers,
                    'body': json.dumps({"message": "Projeto deletado com sucesso"})
                }
            else:
                return {
                    'statusCode': 400,
                    'headers': headers,
                    'body': json.dumps({"error": "ID do projeto é obrigatório para exclusão"})
                }
            
        else:
            return {
                'statusCode': 405,
                'headers': headers,
                'body': json.dumps({"error": "Método não permitido"})
            }
            
    except Exception as e:
        return {
            'statusCode': 500,
            'headers': headers,
            'body': json.dumps({"error": str(e)})
        }